### ISSUE TITLE
An instance of analyzer Microsoft.EntityFrameworkCore.InternalUsageDiagnosticAnalyzer

### ISSUE THREAD

**Issue by vecsq** [dotnet/efcore#29984]
## File a bug

Working with Npgsql.EntityFrameworkCore.PostgreSQL v7.0.1 on Ubuntu 22.04 Dotnet 6.0.111, I'm getting warnings on `dotnet build`.  Since adding the library to my project and seeing the warnings, I've also noticed that the dotnet process crashes occasionally (which didn't happen before).

I had a quick read of https://github.com/npgsql/efcore.pg/issues/1507 and it seems similar.

While the issue is happening on my local dev environment, I've reproduced in a container.  The Dockerfile and build output below shows the issue.  Also reproduceable with `Oracle.EntityFrameworkCore`.  Interestingly enough, a project with `MySql.EntityFrameworkCore` builds just fine.

### Include your code

```Dockerfile
FROM ubuntu:22.04 AS builder

RUN apt-get update && apt-get install -y dotnet6 ca-certificates

WORKDIR /src

RUN dotnet --version

RUN dotnet new console && \
    dotnet add package Npgsql.EntityFrameworkCore.PostgreSQL && \
    dotnet build
```
### Include stack traces

```log
Sending build context to Docker daemon  2.048kB
Step 1/5 : FROM ubuntu:22.04 AS builder
 ---> 6b7dfa7e8fdb
Step 2/5 : RUN apt-get update && apt-get install -y dotnet6 ca-certificates
 ---> Using cache
 ---> 71adca1a3faa
Step 3/5 : WORKDIR /src
 ---> Using cache
 ---> cef8b6eb1a8d
Step 4/5 : RUN dotnet --version
 ---> Using cache
 ---> 23a7dcf315a6
Step 5/5 : RUN dotnet new console &&     dotnet add package Npgsql.EntityFrameworkCore.PostgreSQL &&     dotnet build
 ---> Running in 693b33402a0b

Welcome to .NET 6.0!
---------------------
SDK Version: 6.0.111

----------------
Installed an ASP.NET Core HTTPS development certificate.
To trust the certificate run 'dotnet dev-certs https --trust' (Windows and macOS only).
Learn about HTTPS: https://aka.ms/dotnet-https
----------------
Write your first app: https://aka.ms/dotnet-hello-world
Find out what's new: https://aka.ms/dotnet-whats-new
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
The template "Console App" was created successfully.

Processing post-creation actions...
Running 'dotnet restore' on /src/src.csproj...
  Determining projects to restore...
  Restored /src/src.csproj (in 93 ms).
Restore succeeded.

  Determining projects to restore...
  Writing /tmp/tmpHnQ4qs.tmp
info : Adding PackageReference for package 'Npgsql.EntityFrameworkCore.PostgreSQL' into project '/src/src.csproj'.
info :   GET https://api.nuget.org/v3/registration5-gz-semver2/npgsql.entityframeworkcore.postgresql/index.json
info :   OK https://api.nuget.org/v3/registration5-gz-semver2/npgsql.entityframeworkcore.postgresql/index.json 761ms
info : Restoring packages for /src/src.csproj...
info :   GET https://api.nuget.org/v3-flatcontainer/npgsql.entityframeworkcore.postgresql/index.json
info :   OK https://api.nuget.org/v3-flatcontainer/npgsql.entityframeworkcore.postgresql/index.json 197ms
info :   GET https://api.nuget.org/v3-flatcontainer/npgsql.entityframeworkcore.postgresql/7.0.1/npgsql.entityframeworkcore.postgresql.7.0.1.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/npgsql.entityframeworkcore.postgresql/7.0.1/npgsql.entityframeworkcore.postgresql.7.0.1.nupkg 4ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.abstractions/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.relational/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/npgsql/index.json
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.abstractions/index.json 231ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.abstractions/7.0.1/microsoft.entityframeworkcore.abstractions.7.0.1.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.abstractions/7.0.1/microsoft.entityframeworkcore.abstractions.7.0.1.nupkg 4ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.relational/index.json 272ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.relational/7.0.1/microsoft.entityframeworkcore.relational.7.0.1.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.relational/7.0.1/microsoft.entityframeworkcore.relational.7.0.1.nupkg 4ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.configuration.abstractions/index.json
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore/index.json 784ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore/7.0.1/microsoft.entityframeworkcore.7.0.1.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore/7.0.1/microsoft.entityframeworkcore.7.0.1.nupkg 5ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.analyzers/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.caching.memory/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json
info :   OK https://api.nuget.org/v3-flatcontainer/npgsql/index.json 823ms
info :   GET https://api.nuget.org/v3-flatcontainer/npgsql/7.0.1/npgsql.7.0.1.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/npgsql/7.0.1/npgsql.7.0.1.nupkg 4ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/system.runtime.compilerservices.unsafe/index.json
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.caching.memory/index.json 194ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.caching.memory/7.0.0/microsoft.extensions.caching.memory.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.caching.memory/7.0.0/microsoft.extensions.caching.memory.7.0.0.nupkg 3ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.caching.abstractions/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection.abstractions/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/index.json
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.primitives/index.json
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/index.json 210ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/7.0.0/microsoft.extensions.logging.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging/7.0.0/microsoft.extensions.logging.7.0.0.nupkg 29ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.configuration.abstractions/index.json 757ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.configuration.abstractions/7.0.0/microsoft.extensions.configuration.abstractions.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.configuration.abstractions/7.0.0/microsoft.extensions.configuration.abstractions.7.0.0.nupkg 3ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/index.json 194ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/6.0.0/microsoft.extensions.logging.abstractions.6.0.0.nupkg
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/7.0.0/microsoft.extensions.logging.abstractions.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/6.0.0/microsoft.extensions.logging.abstractions.6.0.0.nupkg 3ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.logging.abstractions/7.0.0/microsoft.extensions.logging.abstractions.7.0.0.nupkg 3ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection.abstractions/index.json 199ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection.abstractions/7.0.0/microsoft.extensions.dependencyinjection.abstractions.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.caching.abstractions/index.json 204ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.caching.abstractions/7.0.0/microsoft.extensions.caching.abstractions.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection.abstractions/7.0.0/microsoft.extensions.dependencyinjection.abstractions.7.0.0.nupkg 5ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.caching.abstractions/7.0.0/microsoft.extensions.caching.abstractions.7.0.0.nupkg 4ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.primitives/index.json 228ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.primitives/7.0.0/microsoft.extensions.primitives.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.primitives/7.0.0/microsoft.extensions.primitives.7.0.0.nupkg 3ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.analyzers/index.json 747ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.analyzers/7.0.1/microsoft.entityframeworkcore.analyzers.7.0.1.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.entityframeworkcore.analyzers/7.0.1/microsoft.entityframeworkcore.analyzers.7.0.1.nupkg 4ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection/index.json 754ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection/7.0.0/microsoft.extensions.dependencyinjection.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.dependencyinjection/7.0.0/microsoft.extensions.dependencyinjection.7.0.0.nupkg 3ms
info :   OK https://api.nuget.org/v3-flatcontainer/system.runtime.compilerservices.unsafe/index.json 762ms
info :   GET https://api.nuget.org/v3-flatcontainer/system.runtime.compilerservices.unsafe/6.0.0/system.runtime.compilerservices.unsafe.6.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/system.runtime.compilerservices.unsafe/6.0.0/system.runtime.compilerservices.unsafe.6.0.0.nupkg 3ms
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/index.json 791ms
info :   GET https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/7.0.0/microsoft.extensions.options.7.0.0.nupkg
info :   OK https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/7.0.0/microsoft.extensions.options.7.0.0.nupkg 3ms
info : Installed Microsoft.EntityFrameworkCore.Analyzers 7.0.1 from https://api.nuget.org/v3/index.json with content hash 4A4NFkPFoRMdry24Hjdb5/sOH3tU3hzFePTGnSg8P+VBrFp/EO8kcA4L0hBzwbXj2HGNL8I/quYyS5GVtb4EOg==.
info : Installed Microsoft.EntityFrameworkCore.Abstractions 7.0.1 from https://api.nuget.org/v3/index.json with content hash pPaoi2RHAkrsJ2kcGZGBiJFUXSS1LC+qET6HCn4hXP+X/kvIriJLv8lx6ddI6gTluGvm/lDgPAA0zZDXYaKf3A==.
info : Installed System.Runtime.CompilerServices.Unsafe 6.0.0 from https://api.nuget.org/v3/index.json with content hash /iUeP3tq1S0XdNNoMz5C9twLSrM/TH+qElHkXWaPvuNOt+99G75NrV0OS2EqHx5wMN7popYjpc8oTjC1y16DLg==.
info : Installed Microsoft.Extensions.Caching.Abstractions 7.0.0 from https://api.nuget.org/v3/index.json with content hash IeimUd0TNbhB4ded3AbgBLQv2SnsiVugDyGV1MvspQFVlA07nDC7Zul7kcwH5jWN3JiTcp/ySE83AIJo8yfKjg==.
info : Installed Microsoft.Extensions.Configuration.Abstractions 7.0.0 from https://api.nuget.org/v3/index.json with content hash f34u2eaqIjNO9YLHBz8rozVZ+TcFiFs0F3r7nUJd7FRkVSxk8u4OpoK226mi49MwexHOR2ibP9MFvRUaLilcQQ==.
info : Installed Microsoft.Extensions.Primitives 7.0.0 from https://api.nuget.org/v3/index.json with content hash um1KU5kxcRp3CNuI8o/GrZtD4AIOXDk+RLsytjZ9QPok3ttLUelLKpilVPuaFT3TFjOhSibUAso0odbOaCDj3Q==.
info : Installed Microsoft.Extensions.Caching.Memory 7.0.0 from https://api.nuget.org/v3/index.json with content hash xpidBs2KCE2gw1JrD0quHE72kvCaI3xFql5/Peb2GRtUuZX+dYPoK/NTdVMiM67Svym0M0Df9A3xyU0FbMQhHw==.
info : Installed Microsoft.EntityFrameworkCore.Relational 7.0.1 from https://api.nuget.org/v3/index.json with content hash F4EYKRhtnkjxp4BhKF3XDn+D3AKok6KIRZYAfaj6knx80CsjFeEUxFpSoqlB9r7ZJTgz30f2oZpHOO4qasIkhA==.
info : Installed Microsoft.Extensions.Logging 7.0.0 from https://api.nuget.org/v3/index.json with content hash Nw2muoNrOG5U5qa2ZekXwudUn2BJcD41e65zwmDHb1fQegTX66UokLWZkJRpqSSHXDOWZ5V0iqhbxOEky91atA==.
info : Installed Microsoft.Extensions.Options 7.0.0 from https://api.nuget.org/v3/index.json with content hash lP1yBnTTU42cKpMozuafbvNtQ7QcBjr/CcK3bYOGEMH55Fjt+iecXjT6chR7vbgCMqy3PG3aNQSZgo/EuY/9qQ==.
info : Installed Microsoft.Extensions.DependencyInjection 7.0.0 from https://api.nuget.org/v3/index.json with content hash elNeOmkeX3eDVG6pYVeV82p29hr+UKDaBhrZyWvWLw/EVZSYEkZlQdkp0V39k/Xehs2Qa0mvoCvkVj3eQxNQ1Q==.
info : Installed Microsoft.EntityFrameworkCore 7.0.1 from https://api.nuget.org/v3/index.json with content hash uvzJkdmqbB50COlCiNaBjWc4cl3kHBO9P7glPk6wK8xyLrli4xeVip+pmMyT/kYc2shWm1YdxegxFqKeCvQXAA==.
info : Installed Microsoft.Extensions.DependencyInjection.Abstractions 7.0.0 from https://api.nuget.org/v3/index.json with content hash h3j/QfmFN4S0w4C2A6X7arXij/M/OVw3uQHSOFxnND4DyAzO1F9eMX7Eti7lU/OkSthEE0WzRsfT/Dmx86jzCw==.
info : Installed Npgsql.EntityFrameworkCore.PostgreSQL 7.0.1 from https://api.nuget.org/v3/index.json with content hash 8txq2CsPWvWO4azZSBmKLKmYsL6dkGrarX2hJZY0Djx6uWZJGH9W0OTE1oHqn+aC5L+xrzyS+TM3kiKw2gdXuA==.
info : Installed Microsoft.Extensions.Logging.Abstractions 6.0.0 from https://api.nuget.org/v3/index.json with content hash /HggWBbTwy8TgebGSX5DBZ24ndhzi93sHUBDvP1IxbZD7FDokYzdAr6+vbWGjw2XAfR2EJ1sfKUotpjHnFWPxA==.
info : Installed Npgsql 7.0.1 from https://api.nuget.org/v3/index.json with content hash +s2gUPA37wDy9B3mE7ZN8xhfnSrf56OCR8ELMZkEbNxk9MyWS54o7ttrngmu7s7/C2ldjAWMdggkpMMPCZsKSg==.
info : Installed Microsoft.Extensions.Logging.Abstractions 7.0.0 from https://api.nuget.org/v3/index.json with content hash kmn78+LPVMOWeITUjIlfxUPDsI0R6G0RkeAMBmQxAJ7vBJn4q2dTva7pWi65ceN5vPGjJ9q/Uae2WKgvfktJAw==.
info : Package 'Npgsql.EntityFrameworkCore.PostgreSQL' is compatible with all the specified frameworks in project '/src/src.csproj'.
info : PackageReference for package 'Npgsql.EntityFrameworkCore.PostgreSQL' version '7.0.1' added to file '/src/src.csproj'.
info : Committing restore...
info : Generating MSBuild file /src/obj/src.csproj.nuget.g.props.
info : Generating MSBuild file /src/obj/src.csproj.nuget.g.targets.
info : Writing assets file to disk. Path: /src/obj/project.assets.json
log  : Restored /src/src.csproj (in 2.28 sec).
Microsoft (R) Build Engine version 17.0.1+b177f8fa7 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
CSC : warning CS8032: An instance of analyzer Microsoft.EntityFrameworkCore.InternalUsageDiagnosticAnalyzer cannot be created from /root/.nuget/packages/microsoft.entityframeworkcore.analyzers/7.0.1/analyzers/dotnet/cs/Microsoft.EntityFrameworkCore.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [/src/src.csproj]
CSC : warning CS8032: An instance of analyzer Microsoft.EntityFrameworkCore.UninitializedDbSetDiagnosticSuppressor cannot be created from /root/.nuget/packages/microsoft.entityframeworkcore.analyzers/7.0.1/analyzers/dotnet/cs/Microsoft.EntityFrameworkCore.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [/src/src.csproj]
  src -> /src/bin/Debug/net6.0/src.dll

Build succeeded.

CSC : warning CS8032: An instance of analyzer Microsoft.EntityFrameworkCore.InternalUsageDiagnosticAnalyzer cannot be created from /root/.nuget/packages/microsoft.entityframeworkcore.analyzers/7.0.1/analyzers/dotnet/cs/Microsoft.EntityFrameworkCore.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [/src/src.csproj]
CSC : warning CS8032: An instance of analyzer Microsoft.EntityFrameworkCore.UninitializedDbSetDiagnosticSuppressor cannot be created from /root/.nuget/packages/microsoft.entityframeworkcore.analyzers/7.0.1/analyzers/dotnet/cs/Microsoft.EntityFrameworkCore.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [/src/src.csproj]
    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.83
Removing intermediate container 693b33402a0b
 ---> a8ef0e79f145
Successfully built a8ef0e79f145
```

### Include provider and version information

EF Core version: 7.0.1
Database provider: Npgsql.EntityFrameworkCore.PostgreSQL, Oracle.EntityFrameworkCore
Target framework: net6.0 
Operating system: Ubuntu 22.04
IDE: none

**Comment by michaeljenzen** [comment]
I am experiencing the same issue on a Linux builder server after upgrading to EF7.  This project is using SQL Server as the DB provider.

**Comment by roji** [comment]
@vecsq @michaeljenzen can you please check if the same warning is reported when targeting net7.0 instead of net6.0?

**Comment by michaeljenzen** [comment]
@roji, due to a pileup of annoying little issues, I cannot, hopefully vecsq can.  I will say though this is a "works on my machine" (net6.0 on Windows) scenario though.

**Comment by michaeljenzen** [comment]
@vecsq, @roji, found the fix.  Upgraded .NET SDK from 6.0.101 (6.0.1) to 6.0.302 (6.0.6) solved this problem for me.

**Comment by vecsq** [comment]
Thanks @michaeljenzen. Manual install of v6.0.404 worked for me.

**Comment by bricelam** [comment]
Re-opening to discuss as a team in Triage.

Version 6.0.1xx of the .NET SDK is still supported. Our analyzer does not work with it (or VS 17.0).

This is because we upgraded the Roslyn packages to 4.2. Here are the minimum versions for each version of the SDK:

SDK | Roslyn
--- | ---
6.0.4xx | 4.3.0
6.0.3xx | 4.2.0
6.0.1xx | 4.0.1

**Comment by bricelam** [comment]
Minimal repro:

```cmd
dotnet new globaljson --sdk-version 6.0.112 --force
dotnet new console --use-program-main --name ConsoleApp --output .
dotnet add ConsoleApp.csproj Microsoft.EntityFrameworkCore.Analyzers
dotnet msbuild -t:rebuild -bl
```

```
CSC : warning CS8032: An instance of analyzer Microsoft.EntityFrameworkCore.InternalUsageDiagnosticAnalyzer cannot be created from C:\Users\pamorgad\.nuget\packages\microsoft.entityframeworkcore.analyzers\7.0.1\analyzers\dotnet\cs\Microsoft.EntityFrameworkCore.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [C:\Temp\VSL\2208240030002590\2023-01-09\repro\ConsoleApp.csproj]
CSC : warning CS8032: An instance of analyzer Microsoft.EntityFrameworkCore.UninitializedDbSetDiagnosticSuppressor cannot be created from C:\Users\pamorgad\.nuget\packages\microsoft.entityframeworkcore.analyzers\7.0.1\analyzers\dotnet\cs\Microsoft.EntityFrameworkCore.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [C:\Temp\VSL\2208240030002590\2023-01-09\repro\ConsoleApp.csproj]
```

**Comment by ajcvickers** [comment]
@bricelam I think people are supposed to always use the latest SDK, even if they are targeting an earlier runtime.

**Comment by vecsq** [comment]
Shouldn't the process of adding a NuGet package fail if the a dependency or runtime is not supported?

**Comment by ajcvickers** [comment]
Notes from triage: First course of action is to update VS; @bricelam to suggest this on internal thread. If this isn't possible, then attempt workaround such as disabling the analyzer or forcing/accepting a downgrade of the version.

**Comment by khhsenti** [comment]
So this is a bit of a fun one to stumble into when using Ubuntu 24.04 LTS :)

The recommended way to install the dotnet 6 SDK only contains the 6.0.1xx versions, not any of the other "branches". The official ubuntu repo contains only 8.0 SDK, the backport only contains 6.0.1xx and the MS repo does not support 24.04. It is possible to install it manually of course which as I understand fixes this issue.

I tried searching for figuring out the reason for these multiple active "branches", could not find it. Seems weird to not have the latest one available, but this is not really an EFcore problem either i guess :)

This is not a complaint, i understand why this is closed as fixed I am just letting you know it hits all 24.04 users currently.

**Comment by roji** [comment]
@khhsenti thanks for that added details. Some more info on dotnet SDK versioning ("branches") can be found [here](https://learn.microsoft.com/en-us/dotnet/core/versions/).
